pkg/deploy/lattice/access_log_subscription_synthesizer.go

```diff
@@ -40,25 +38,32 @@ func (s *accessLogSubscriptionSynthesizer) Synthesize(ctx context.Context) error
 		return err
 	}
 
-	var errs []string
 	for _, als := range accessLogSubscriptions {
-		if !als.Spec.IsDeleted {
-			s.log.Debugf("Started creating or updating access log subscription %s", als.ID())
-			_, err := s.accessLogSubscriptionManager.Create(ctx, als)
+		switch als.Spec.EventType {
+		case core.CreateEvent:
+			s.log.Debugf("Started creating Access Log Subscription %s", als.ID())
+			alsStatus, err := s.accessLogSubscriptionManager.Create(ctx, als)
 			if err != nil {
-				errMsg := fmt.Sprintf("synthesizing access log subscription %s failed due to: %s", als.ID(), err)
-				errs = append(errs, errMsg)
+				return err
 			}
-		} else {
-			s.log.Debugf("Started deleting access log subscription %s", als.ID())
+			als.Status = alsStatus
+		case core.UpdateEvent:
+			s.log.Debugf("Started updating Access Log Subscription %s", als.ID())
 			// TODO
+			return nil
+		case core.DeleteEvent:
+			s.log.Debugf("Started deleting Access Log Subscription %s", als.ID())
+			if als.Status != nil {
+				err := s.accessLogSubscriptionManager.Delete(ctx, als)
+				if err != nil {
+					return err
+				}
+			} else {
+				s.log.Debugf("Ignoring deletion of Access Log Subscription because als %s has no ARN", als.ID())
+			}
```

Comment:
nit: try to avoid nested code

```go

s.log.Debugf("Started deleting Access Log Subscription %s", als.ID())
if als.Status == nil {
	s.log.Debugf("Ignoring deletion of Access Log Subscription because als %s has no ARN", als.ID())
	return nil
}
err := s.accessLogSubscriptionManager.Delete(ctx, als)
if err != nil {
	return err
}
```

Reply:
Fixed next commit!